<launch>
  <!--Set Parameters-->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="is_sim" default="true" doc="true if running in simulator, false if running on real Turtlebot hardware"/>
  <arg name="x_pos" default="-2.2"/>
  <arg name="y_pos" default="0.4"/>
  <arg name="z_pos" default="0.05"/>

  <env name="GAZEBO_MODEL_PATH" value="$(find rbe3002_final)/models:$(optenv GAZEBO_MODEL_PATH"/>
  
  <!-- If is_sim, run the simulator! -->
  <group if="$(arg is_sim)">
    <include file="$(find rbe3002-turtlebot)/launch/turtlebot3002_world.launch">
      <arg name="world_name" value="$(find rbe3002_final)/worlds/final_maze.world"/>
      <arg name="x_pos" value="$(arg x_pos)"/>
      <arg name="y_pos" value="$(arg y_pos)"/>
      <arg name="z_pos" value="$(arg z_pos)"/>
    </include>
  </group>
  
  <!-- If not is_sim, run the bringup for a turltebot3 -->
  <group unless="$(arg is_sim)">
    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch" />
  </group>

  <!--gmapping. We don't need mapserver if gmapping is running -->
  <!--NOTE: Set opeb_rviz config to false to use our config file -->
  <include file="$(find turtlebot3_slam)/launch/turtlebot3_slam.launch" >
    <arg name='open_rviz' value='False'/>
  </include>

  <!-- AMCL -->
  <include file="$(find turtlebot3_navigation)/launch/amcl.launch"/>

  <!--Run the tranformer to set the center of the odom frame to the map frame-->
  <node name="static_transform_publisher" type="static_transform_publisher" pkg="tf" args="0 0 0 0 0 0 /map /odom 100"/>

  <!--Launch RViz with custom config file used for lab4-->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find rbe3002_lab3)/launch/turtlebot3_gazebo_model_lab4a.rviz"/>

  <!--Launch Path Planner Node for lab 4-->
  <node name='path_planner' pkg='rbe3002_lab3' type='path_planner_lab4.py'/>

  <!--TODO: Launch motion node and any other required ones-->
  <!--TODO: Configure nodes for path planning and the lab 2 motion node-->

</launch>